pkg/csource/csource.go

```diff
@@ -15,6 +34,7 @@ import (
 	"github.com/google/syzkaller/sys/targets"
 )
 
+// Write generates C source for program p based on the provided options opt.
```

Comment:
I think no in-depth comments are required yet (if it's not trivial to add). But *interface* descriptions help, where the inputs and outputs aren't obvious. E.g. it's not obvious that `[]byte` is a C source program, but we can guess.

Are there non-trivial pre- and post-conditions (a'la "Requires: ...", "Ensures: ...")?

Similar for the helper functions. E.g. I cannot tell at a glance what the function generateSyscalls with a signature of `([]string, hasVars bool) -> string` will do for me.